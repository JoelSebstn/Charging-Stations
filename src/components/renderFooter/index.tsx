import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { useNavigation } from "@react-navigation/native";


export function renderFooter(isLoading: any) {

    // Check If Loading
    if (isLoading) {
        return (
            <ActivityIndicator />
        )
    }
    else {
        return null;
    }

};