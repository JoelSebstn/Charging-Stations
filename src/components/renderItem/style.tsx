import { StyleSheet } from "react-native";

export const Styles = StyleSheet.create({
    alignCentre: { flex: 1, alignItems: 'flex-start', marginLeft: 8 },
    flexRow: { flexDirection: 'row' },
    textFont22: { fontSize: 22 },
    rowMargin12:
        { flexDirection: 'row', marginVertical: 12 }
});