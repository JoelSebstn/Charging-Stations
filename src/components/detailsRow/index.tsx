import { View, Text } from "react-native";
import { Styles } from "./style";

export function DetailsRow({ rowName, data }: any) {
    return (
        <View style={[Styles.flexRow, Styles.margin20]}>
            <View style={Styles.alignStart}><Text style={Styles.textFont22}>{rowName}:</Text></View>
            <View style={Styles.alignStart}><Text style={Styles.textFont22}> {data}</Text></View>
        </View >)
}