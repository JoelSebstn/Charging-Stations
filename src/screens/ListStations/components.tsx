import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { Styles } from "./style";
import { useNavigation } from "@react-navigation/native";

export function HeaderRow() {
    return (<><View style={Styles.flexRow}>
        <View style={Styles.alignCentre}><Text style={Styles.textFont22}>Name</Text></View>
        <View style={Styles.alignCentre}><Text style={Styles.textFont22}> Status</Text></View>
    </View>
        <View
            style={Styles.horizontalLine}
        /></>
    )
}
export function RenderItem(item: any) {
    const data = item["item"].data();
    const stationId = item["item"].data().id;
    const navigation: any = useNavigation();
    return (
        <TouchableOpacity onPress={() => navigation.navigate('DetailsSCene', { stationId: stationId })}>
            <View style={{ flexDirection: 'row', marginVertical: 12 }}>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}>{data.name}</Text></View>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}> {data.status}</Text></View>
            </View>
        </TouchableOpacity>
    )

}
export function renderFooter(isLoading: any) {
    try {
        // Check If Loading
        if (isLoading) {
            return (
                <ActivityIndicator />
            )
        }
        else {
            return null;
        }
    }
    catch (error) {
        console.log(error);
    }
};