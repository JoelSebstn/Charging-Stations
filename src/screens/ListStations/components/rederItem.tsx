import { useNavigation } from "@react-navigation/native";
import { TouchableOpacity, View, Text } from "react-native";
import { Styles } from "../style";

export function RenderItem(item: any) {
    const data = item["item"];
    const stationId = data.id;
    const navigation: any = useNavigation();
    return (
        <TouchableOpacity onPress={() => navigation.navigate('DetailsSCene', { stationId: stationId })}>
            <View style={Styles.rowMargin12}>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}>{data.name}</Text></View>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}> {data.status}</Text></View>
            </View>
        </TouchableOpacity>
    )

}