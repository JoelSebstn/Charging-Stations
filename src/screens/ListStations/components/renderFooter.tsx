import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native";
import { Styles } from "../style";
import { useNavigation } from "@react-navigation/native";


export function renderFooter(isLoading: any) {
    try {
        // Check If Loading
        if (isLoading) {
            return (
                <ActivityIndicator />
            )
        }
        else {
            return null;
        }
    }
    catch (error) {
        console.log(error);
    }
};