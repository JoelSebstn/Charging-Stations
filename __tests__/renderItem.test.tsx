import { fireEvent, render } from "@testing-library/react-native";
import { RenderItem } from "../src/components/renderItem";
import { NavigationContainer } from "@react-navigation/native";
import renderer from 'react-test-renderer';
const mockData = { "id": 123, "name": "Harry", "status": "Available" }
describe('Testing rendering Item', () => {
    it('Check status contents', () => {
        const { getByText } = render(<NavigationContainer><RenderItem item={mockData} /></NavigationContainer>)
        const label = getByText("Available");
        expect(label).toBeTruthy();
    })
    it('should render name', () => {
        const { getByText } = render(
            <NavigationContainer><RenderItem item={mockData} /></NavigationContainer>,
        );
        const label = getByText("Harry");
        expect(label).toBeTruthy();
    });
    it("Matches Header Snapshot", () => {
        const domTree = renderer.create(<NavigationContainer><RenderItem item={mockData} /></NavigationContainer>).toJSON();
        expect(domTree).toMatchSnapshot();
    });
    it('check click', () => {
        const { getByText } = render(
            <NavigationContainer><RenderItem item={mockData} /></NavigationContainer>,
        );
        fireEvent.press(getByText("Harry"));
        expect(getByText('Available')).toBeTruthy();
    })
})