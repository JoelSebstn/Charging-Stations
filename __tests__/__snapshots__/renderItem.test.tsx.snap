// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering Item Matches Header Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 12,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Harry
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
         
        Available
      </Text>
    </View>
  </View>
</View>
`;
